src/test/scala/ApplicationSpec.scala

```diff
@@ -0,0 +1,55 @@
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.Source
+import org.json.JSONObject
+
+class ApplicationSpec extends AnyFlatSpec with Matchers {
+
+  "A successful xml" should "pass simply" in {
+    val xmlTestURL = getClass.getResource("/GradeSchool_successful.xml")
+    val jsonArray = Application.getTestCasesJSON(xmlTestURL)
+    val objects = (0 until jsonArray.length).map( jsonArray.getJSONObject(_).optJSONObject("failure") )
+    objects should contain only (null)
+  }
+
+  "A successful xml" should "be properly formatted as JSON" in {
+    val xmlTestURL = getClass.getResource("/GradeSchool_successful.xml")
+    val outputFileURL = getClass.getResource("/outputs/output.txt")
+    val exercismOutput: JSONObject = Application.toExercismJSON(xmlTestURL, outputFileURL)
+    assert(exercismOutput.getInt("version") == 2)
+    assert(exercismOutput.getString("status") == "pass")
+    assert(exercismOutput.getString("message") == "")
+    assert(exercismOutput.get("tests").asInstanceOf[Array[JSONObject]].length == 7)
```

Comment:
I would suggest also verifying the exact contents of the `tests` field, to verify that the keys of each element in the `tests` array has the proper value (most importantly the `name` and the `status`). This comment also applies to the other tests.

Reply:
I added a new commit, is that what you meant? I thought I could assert field by field, when I add also `output` and `test_case`, in a second PR.

Reply:
Yeah sure. I like how the new test verifies the contents.